appinventor/components/src/com/google/appinventor/components/runtime/util/FileUtil.java

```diff
@@ -971,7 +971,9 @@ public static boolean isAppSpecificExternalUri(Form form, String fileUri) {
    */
   @SuppressWarnings({"unused", "deprecation"})
   public static boolean isExternalStorageUri(Form form, String fileUri) {
-    if (fileUri.startsWith("file:///sdcard/") || fileUri.startsWith("file:///storage")) {
+    if (fileUri.startsWith("file:///sdcard/")
+            || fileUri.startsWith("file:///storage")
+            || fileUri.startsWith("file:/storage")) {
```

Comment:
Indentation only needs to be +4 here from previous line.

Separately, under what conditions do we end up with "file:/storage" rather than "file:///storage"? And can this yield the path "file:/sdcard"?

Reply:
The RUtil calls the `isExternalStorageUri` in the FileUtil.
Suppose we call the RUtil with "/storage/emulated/0/..", it just appends "file:" it removed previously, thus forming path "file:/storage/emulated/..", the FileUtil only checked for starting with "file:///storage" before, i added a condition to check paths starting from file:/storage/..